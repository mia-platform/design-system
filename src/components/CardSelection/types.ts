
export enum Type {
  Radio = 'radio',
  Checkbox = 'checkbox'
}
