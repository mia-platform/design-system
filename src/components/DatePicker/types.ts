import {
  DisabledDate as AntDisabledDate,
} from 'rc-picker/lib/interface'
import { Dayjs } from 'dayjs'

export type DisabledDate = AntDisabledDate<Dayjs>
export enum ShowTimeOptions {
  Hours = 'hours',
  Minutes = 'minutes',
  Seconds = 'seconds'
}

export type ShowTime = boolean | {
  showHour: boolean;
  showMinute: boolean;
  showSecond: boolean;
}
