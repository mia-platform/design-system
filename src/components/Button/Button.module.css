.button {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: calc(var(--spacing-gap-xs, 4) * 1px);
}

/* TODO: add 12px padding to theme */
.buttonSm {
  padding-top: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
  padding-right: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
  padding-bottom: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
  padding-left: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
}

.buttonSmIconOnly {
  padding: calc(var(--spacing-padding-xs, 4) * 1px - 1px) !important;
}

.buttonMd {
  padding-top: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
  padding-right: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
  padding-bottom: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
  padding-left: calc((var(--spacing-padding-xs, 4) + var(--spacing-padding-sm, 8)) * 1px - 1px) !important;
}

.buttonMdIconOnly {
  padding: calc(var(--spacing-padding-sm, 8) * 1px - 1px) !important;
}

.buttonLg {
  height: calc(var(--shape-size-4xl, 48) * 1px) !important;
  padding: calc(var(--spacing-padding-md, 16) * 1px - 1px) !important;
}

.buttonGhost {
  border-color: transparent !important;
  background-color: transparent;
}

.buttonText {
  padding: 0 calc(var(--spacing-padding-xs, 4) * 1px);
}
