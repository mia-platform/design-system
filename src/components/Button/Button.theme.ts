/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ComponentsTheme, parse } from '../ThemeProvider/Ant'
import Theme from '../../themes/schema'

/**
 * Generates a Ant theme configuration for Button component based on a theme configuration.
 *
 * @link https://ant.design/components/button#design-token
 *
 * @param {Partial<Theme>} theme - theme configuration.
 * @returns {Partial<ComponentsTheme>} The generated Button Ant theme configuration.
 */
export default ({ typography }: Partial<Theme>): ComponentsTheme['Button'] => ({
  contentFontSizeSM: parse(typography?.action?.fontSize),
  contentFontSize: parse(typography?.action?.fontSize),
  contentFontSizeLG: parse(typography?.action?.fontSize),
})
