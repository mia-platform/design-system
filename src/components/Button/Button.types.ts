export enum ButtonHierarchies {
  Primary = 'primary',
  Neutral = 'neutral',
  Danger = 'danger',
}

export enum ButtonIconPositions {
  Left = 'left',
  Right = 'right',
}

export enum ButtonShapes {
  Square = 'square',
  Circle = 'circle',
}

export enum ButtonSizes {
  Small = 'small',
  Middle = 'middle',
  Large = 'large',
}

export enum ButtonTypes {
  Filled = 'filled',
  Outlined = 'outlined',
  Ghost = 'ghost',
}
