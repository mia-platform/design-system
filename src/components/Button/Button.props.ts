/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { MouseEvent, ReactNode } from 'react'

import { Hierarchy, IconPosition, Shape, Size, Type } from './Button.types'

export type ButtonProps = {

  /**
   * The children nodes to be rendered within the button context.
   */
  children?: ReactNode,

  /**
   * Defines the button hierarchy. Either:
   *
   * - "primary": button associated with the most significant (and therefore primary) action on the page;
   * - "neutral": button associated with a general purpose action;
   * - "danger": button associated with a potentially dangerous action.
   */
  hierarchy?: Hierarchy,

  /**
   * Redirect url of a link button.
   * For security reasons, the attribute "rel: 'noopener noreferrer'" is always specified.
   */
  href?: string,

  /**
   * Sets an icon for the button component.
   */
  icon?: ReactNode,

  /**
   * Defines a position for the button icon (if any). Either:
   *
   * - "left": the icon is placed to the left of the button text;
   * - "right": the icon is placed to the right of the button text.
   */
  iconPosition?: IconPosition,

  /**
   * Identifies whether the button is disabled or not.
   */
  isDisabled?: boolean,

  /**
   * Sets the loading status of the button.
   */
  isLoading?: boolean,

  /**
   * Sets the handler to handle a click event.
   */
  onClick?: (event: MouseEvent) => void,

  /**
   * Defines the button shape. Either:
   *
   * - "square": square-shaped button with rounded corners;
   * - "circle": fully rounded button, typically used for buttons containing only icon.
   */
  shape?: Shape,

  /**
   * Defines the button size. Either:
   *
   * - "small": small button - 24px in height by default;
   * - "middle": middle button - 32px in height by default;
   * - "large": large button - 48px in height by default.
   */
  size?: Size,

  /**
   * Specifies where the linked document will open when the link is clicked.
   * Only usable for link buttons together with the href property.
   *
   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
   */
  target?: string,

  /**
   * Defines the button type. Either:
   *
   * - "filled": button with a solid fill, typically used for primary actions;
   * - "outlined": medium-emphasis button with a stroke around the button container and no fill,
   *   typically used for secondary actions;
   * - "ghost": transparent button with no stroke, typically used for links.
   */
  type?: Type,
}
