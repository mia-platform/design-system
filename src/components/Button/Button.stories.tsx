/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import type { Meta, StoryObj } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
import { Button } from '.'
import { Icon } from '../Icon'

const { Primary, Neutral, Danger } = ButtonHierarchies
const { Left, Right } = ButtonIconPositions
const { Square, Circle } = ButtonShapes
const { Small, Middle, Large } = ButtonSizes
const { Filled, Outlined, Ghost } = ButtonTypes

const icon = <Icon color="white" name="PiCircleHalfTiltLight" size={16} />

const meta = {
  component: Button,
  args: {
    ...Button.defaultProps,
    children: 'Button',
    onClick: action('click'),
  },
} satisfies Meta<typeof Button>

export default meta
type Story = StoryObj<typeof meta>

export const PrimaryFilled: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Filled,
  },
}

export const PrimaryOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Outlined,
  },
}

export const PrimaryGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Primary,
    type: Ghost,
  },
}

export const NeutralOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Neutral,
    type: Outlined,
  },
}

export const NeutralGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Neutral,
    type: Ghost,
  },
}

export const DangerFilled: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Filled,
  },
}

export const DangerOutline: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Outlined,
  },
}

export const DangerGhost: Story = {
  args: {
    ...meta.args,
    hierarchy: Danger,
    type: Ghost,
  },
}

export const SquareShape: Story = {
  args: {
    ...meta.args,
    children: undefined,
    icon,
    shape: Square,
  },
}

export const CircleShape: Story = {
  args: {
    ...meta.args,
    children: undefined,
    icon,
    shape: Circle,
  },
}

export const SmallSize: Story = {
  args: {
    ...meta.args,
    size: Small,
  },
}

export const MiddleSize: Story = {
  args: {
    ...meta.args,
    size: Middle,
  },
}

export const LargeSize: Story = {
  args: {
    ...meta.args,
    size: Large,
  },
}

export const Disabled: Story = {
  args: {
    ...meta.args,
    isDisabled: true,
  },
}

export const Loading: Story = {
  args: {
    ...meta.args,
    isLoading: true,
  },
}

export const WithIconLeft: Story = {
  args: {
    ...meta.args,
    icon,
    iconPosition: Left,
  },
}

export const WithIconRight: Story = {
  args: {
    ...meta.args,
    icon,
    iconPosition: Right,
  },
}
