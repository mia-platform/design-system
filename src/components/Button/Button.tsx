/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, ReactNode, useMemo } from 'react'
import { Button as AntButton } from 'antd'
import classnames from 'classnames'

import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
import styles from './Button.module.css'

const { button, buttonSm, buttonSmIconOnly, buttonMd, buttonMdIconOnly, buttonLg, buttonGhost, buttonText } = styles

const { Primary, Neutral, Danger } = ButtonHierarchies
const { Left, Right } = ButtonIconPositions
const { Square } = ButtonShapes
const { Small, Middle, Large } = ButtonSizes
const { Filled, Ghost } = ButtonTypes

export type ButtonProps = {

  /**
   * The children nodes to be rendered within the button context.
   */
  children?: ReactNode,

  /**
   * Defines the button hierarchy. Either:
   *
   * - primary: button associated with the most significant (and therefore primary) action on the page;
   * - neutral: button associated with a general purpose action;
   * - danger: button associated with a potentially dangerous action.
   */
  hierarchy?: ButtonHierarchies,

  /**
   * Redirect url of a link button.
   * For security reasons, the attribute "rel: 'noopener noreferrer'" is always specified.
   */
  href?: string,

  /**
   * Sets an icon for the button component.
   */
  icon?: ReactNode,

  /**
   * Defines a position for the button icon (if any). Either:
   *
   * - left: the icon is placed to the left of the button text;
   * - right: the icon is placed to the right of the button text.
   */
  iconPosition?: ButtonIconPositions,

  /**
   * Identifies whether the button is disabled or not.
   */
  isDisabled?: boolean,

  /**
   * Sets the loading status of the button.
   */
  isLoading?: boolean,

  /**
   * Sets the handler to handle a click event.
   */
  onClick?: () => void,

  /**
   * Defines the button shape. Either:
   *
   * - square: square-shaped button with rounded corners;
   * - circle: fully rounded button, typically used for buttons containing only icon.
   */
  shape?: ButtonShapes,

  /**
   * Defines the button size. Either:
   *
   * - small: small button - 24px in height by default;
   * - middle: middle button - 32px in height by default;
   * - large: large button - 48px in height by default.
   */
  size?: ButtonSizes,

  /**
   * Specifies where the linked document will open when the link is clicked.
   * Only usable for link buttons together with the href property.
   *
   * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
   */
  target?: string,

  /**
   * Defines the button type. Either:
   *
   * - filled: button with a solid fill, typically used for primary actions;
   * - outlined: medium-emphasis button with a stroke around the button container and no fill,
   *   typically used for secondary actions;
   * - ghost: transparent button with no stroke, typically used for links.
   */
  type?: ButtonTypes,
}

/**
 * UI component for performing actions on the page interacting through clicks
 *
 * @link https://ant.design/components/button
 * @returns {Button} Button component
 */
export const Button = ({
  children,
  hierarchy,
  href,
  icon,
  iconPosition,
  isDisabled,
  isLoading,
  onClick,
  shape,
  size,
  target,
  type,
}: ButtonProps): ReactElement => {
  const buttonClassNames = useMemo(() => classnames(
    [
      button,
      children && size === Small && buttonSm,
      !children && size === Small && buttonSmIconOnly,
      children && size === Middle && buttonMd,
      !children && size === Middle && buttonMdIconOnly,
      size === Large && buttonLg,
      type === Ghost && buttonGhost,
    ]
  ), [children, size, type])

  return (
    <AntButton
      className={buttonClassNames}
      danger={hierarchy === Danger}
      disabled={isDisabled}
      ghost={type !== Filled && hierarchy !== Neutral}
      loading={isLoading}
      shape={shape === Square ? 'default' : 'circle'}
      size={size}
      type={hierarchy === Neutral ? 'default' : 'primary'}
      onClick={onClick}
      {...href && { href, rel: 'noopener noreferrer', target }}
    >
      {iconPosition === Left && icon}
      {children && <div className={buttonText}>{children}</div>}
      {iconPosition === Right && icon}
    </AntButton>
  )
}

Button.defaultProps = {
  hierarchy: Primary,
  iconPosition: Left,
  isDisabled: false,
  isLoading: false,
  shape: Square,
  size: Middle,
  type: Filled,
}
