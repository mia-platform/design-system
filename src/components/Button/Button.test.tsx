import { fireEvent, render, screen } from '@testing-library/react'

import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
import { Button } from '.'
import { Icon } from '../Icon'

const { Neutral, Danger } = ButtonHierarchies
const { Right } = ButtonIconPositions
const { Circle } = ButtonShapes
const { Small, Large } = ButtonSizes
const { Outlined, Ghost } = ButtonTypes

const icon = <Icon color="white" name="PiCircleHalfTiltLight" size={16} />

describe('Button Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders primary filled button correctly', () => {
    const { asFragment } = render(<Button>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders primary outline button correctly', () => {
    const { asFragment } = render(<Button type={Outlined}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders primary ghost button correctly', () => {
    const { asFragment } = render(<Button type={Ghost}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders neutral outline button correctly', () => {
    const { asFragment } = render(<Button hierarchy={Neutral} type={Outlined}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders neutral ghost button correctly', () => {
    const { asFragment } = render(<Button hierarchy={Neutral} type={Ghost}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders danger filled button correctly', () => {
    const { asFragment } = render(<Button hierarchy={Danger}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders danger outline button correctly', () => {
    const { asFragment } = render(<Button hierarchy={Danger} type={Outlined}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders danger ghost button correctly', () => {
    const { asFragment } = render(<Button hierarchy={Danger} type={Ghost}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders square button correctly', () => {
    const { asFragment } = render(<Button icon={icon} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders circle button correctly', () => {
    const { asFragment } = render(<Button icon={icon} shape={Circle} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders small button correctly', () => {
    const { asFragment } = render(<Button size={Small}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders middle button correctly', () => {
    const { asFragment } = render(<Button>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders large button correctly', () => {
    const { asFragment } = render(<Button size={Large}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders disabled button correctly', () => {
    const { asFragment } = render(<Button isDisabled>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()

    const button = screen.getByRole('button', { name: /Button/i })
    expect(button).toBeDisabled()
  })

  test('renders loading button correctly', () => {
    const { asFragment } = render(<Button isLoading>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders button with icon left correctly', () => {
    const { asFragment } = render(<Button icon={icon}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders button with icon right correctly', () => {
    const { asFragment } = render(<Button icon={icon} iconPosition={Right}>{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders button with href correctly', () => {
    const { asFragment } = render(<Button href="https://mia-platform.eu">{'Button'}</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('calls onClick correctly', () => {
    const onClick = jest.fn()

    render(<Button onClick={onClick}>{'Button'}</Button>)

    const button = screen.getByRole('button', { name: /Button/i })
    fireEvent.click(button)
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
