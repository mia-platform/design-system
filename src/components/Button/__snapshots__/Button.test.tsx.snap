// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button with href correctly 1`] = `
<DocumentFragment>
  <a
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    href="https://mia-platform.eu"
    rel="noopener noreferrer"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </a>
</DocumentFragment>
`;

exports[`Button Component renders button with icon left correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    type="button"
  >
    <svg
      alt="PiCircleHalfTiltLight"
      aria-label="PiCircleHalfTiltLight"
      color="white"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
      />
    </svg>
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders button with icon right correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
    <svg
      alt="PiCircleHalfTiltLight"
      aria-label="PiCircleHalfTiltLight"
      color="white"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders circle button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-circle mia-platform-btn-primary button buttonMdIconOnly"
    type="button"
  >
    <svg
      alt="PiCircleHalfTiltLight"
      aria-label="PiCircleHalfTiltLight"
      color="white"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders danger filled button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-dangerous button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders danger ghost button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-background-ghost mia-platform-btn-dangerous button buttonMd buttonGhost"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders danger outline button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-background-ghost mia-platform-btn-dangerous button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders disabled button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    disabled=""
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders large button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-lg button buttonLg"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders loading button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-loading button buttonMd"
    type="button"
  >
    <span
      class="mia-platform-btn-icon mia-platform-btn-loading-icon"
    >
      <span
        aria-label="loading"
        class="anticon anticon-loading anticon-spin"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </span>
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders middle button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders neutral ghost button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-default button buttonMd buttonGhost"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders neutral outline button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-default button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders primary filled button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders primary ghost button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-background-ghost button buttonMd buttonGhost"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders primary outline button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-background-ghost button buttonMd"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders small button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary mia-platform-btn-sm button buttonSm"
    type="button"
  >
    <div
      class="buttonText"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button Component renders square button correctly 1`] = `
<DocumentFragment>
  <button
    class="mia-platform-btn mia-platform-btn-primary button buttonMdIconOnly"
    type="button"
  >
    <svg
      alt="PiCircleHalfTiltLight"
      aria-label="PiCircleHalfTiltLight"
      color="white"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: white;"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
